packages/flutter_tools/test/general.shard/ios/simulators_test.dart

```diff
@@ -535,6 +535,44 @@ void main() {
         osx.environment['IOS_SIMULATOR_LOG_FILE_PATH'] = syslog.path;
       });
 
+      testUsingContext('simulator can parse Xcode 8/iOS 10-style logs', () async {
+        when(mockProcessManager.start(any, environment: null, workingDirectory: null))
+          .thenAnswer((Invocation invocation) {
+          final Process mockProcess = MockProcess();
+          when(mockProcess.stdout)
+            .thenAnswer((Invocation invocation) {
+            return Stream<List<int>>.fromIterable(<List<int>>['''
+Dec 20 17:04:32 md32-11-vm1 Runner[88374]: flutter: Observatory listening on http://127.0.0.1:64213/1Uoeu523990=/'''
+              .codeUnits]);
+          });
+          when(mockProcess.stderr)
+            .thenAnswer((Invocation invocation) => const Stream<List<int>>.empty());
+          // Delay return of exitCode until after stdout stream data, since it terminates the logger.
+          when(mockProcess.exitCode)
```

Comment:
Instead of setting up a potential future timing issue, you could structure the test so that you can hold the exit until you have verified the log output:


```
var completer = Completer<int>();
when(mockProcess.exitCode).thenAnswer((_) {
  return completer.future;
});
...
final DeviceLogReader logReader = device.getLogReader(
  app: await BuildableIOSApp.fromProject(mockIosProject),
 );

await expectLater(logReader.logLines, emitsThrough(...));
completer.complete(0); // not sure if its actually necessary to exit, cloud dispose instead?
```